docs/components/KRate.md

```diff
@@ -0,0 +1,101 @@
+---
+title: KRate
+lang: en-US
+---
+
+# KRate
+
+Used for rating.
+
+## Install
+
+::: code-group
+
+```bash [pnpm]
+pnpm add @ikun-ui/rate
+```
+
+```bash [yarn]
+yarn add @ikun-ui/rate
+```
+
+```bash [npm]
+npm install @ikun-ui/rate
+```
+
+:::
+
+## Basic usage
+
+Use `colors` prop to set color for different level.
+
+<demo src="../../../../example/rate/basic.svelte"  github='Rate'></demo>
+
+## With allowHalf
+
+Add attribute `allowHalf` Half star allowed.
+
+<demo src="../../../../example/rate/allow-half.svelte"  github='Rate'></demo>
+
+## With text
+
+Using text to indicate rating score.
+
+Add attribute `showText` to display text at the right of Rate. You can assign `texts` for different scores using texts.
+
+<demo src="../../../../example/rate/with-text.svelte" github='Rate'></demo>
+
+## ShowScore and ScoreTemplate
+
+Add attribute `showScore` to display score at the right of Rate. You can customize score display by set `scoreTemplate`, which is a string and internally uses `{value}` to represent the score.
+
+<demo src="../../../../example/rate/score-template.svelte" github='Rate'></demo>
+
+## Clearable
+
+You can reset the value to `0` when you click at the same value again.
+
+<demo src="../../../../example/rate/clearable.svelte" github='Rate'></demo>
+
+## More icons
+
+You can use different icons to distinguish different rate components.
+
+You can customize `icons` by passing icons.
+
+<demo src="../../../../example/rate/more-icon.svelte" github='Rate'></demo>
+
+## Readonly
```

Comment:
I think read only and disabled are different.
In the example you used disabled to implement read-only 🤔

Reply:
> I think read only and disabled are different. In the example you used disabled to implement read-only 🤔

I currently don't have a good idea about the difference between `read-only` and `disabled`. Maybe I can first change the `Readonly` part of the document to `Disabled`.